### ISSUE TITLE
Problem about nodule classification using DPN

### ISSUE THREAD

**Issue by TracelessLe** [wentaozhu/DeepLung#28]
Recently, I try to run the code you provided about nodule classification using DPN named "main_nodcls.py". But it is strange that the acc and loss (~0.693) does not change around epoch 100, and the acc is low.
The config is:
`

    neptime = 2
    def get_lr(epoch):
        if epoch < 150 * neptime:
            lr = 0.1
        elif epoch < 250 * neptime:
            lr = 0.01
        else:
            lr = 0.001
        return lr
`

The train log is:
`

Epoch: 136
INFO:root:ep 136 tracc 0.560175054705 lr 0.1 gbtacc 0.890590809628
INFO:root:Saving..
INFO:root:teacc 46.6666666667 bestacc 57.7777777778 ccgbt 0.844444444444 bestgbt 0.888888888889
INFO:root:
Epoch: 137
INFO:root:ep 137 tracc 0.560175054705 lr 0.1 gbtacc 0.890590809628
INFO:root:Saving..
INFO:root:teacc 46.6666666667 bestacc 57.7777777778 ccgbt 0.844444444444 bestgbt 0.888888888889
INFO:root:
Epoch: 138
INFO:root:ep 138 tracc 0.560175054705 lr 0.1 gbtacc 0.890590809628
INFO:root:Saving..
INFO:root:teacc 46.6666666667 bestacc 57.7777777778 ccgbt 0.844444444444 bestgbt 0.888888888889
INFO:root:
Epoch: 139
INFO:root:ep 139 tracc 0.560175054705 lr 0.1 gbtacc 0.890590809628
INFO:root:Saving..
INFO:root:teacc 46.6666666667 bestacc 57.7777777778 ccgbt 0.844444444444 bestgbt 0.888888888889
INFO:root:
Epoch: 140
INFO:root:ep 140 tracc 0.560175054705 lr 0.1 gbtacc 0.890590809628
INFO:root:Saving..
INFO:root:teacc 46.6666666667 bestacc 57.7777777778 ccgbt 0.844444444444 bestgbt 0.888888888889
INFO:root:
Epoch: 141
INFO:root:ep 141 tracc 0.560175054705 lr 0.1 gbtacc 0.890590809628
INFO:root:Saving..
INFO:root:teacc 46.6666666667 bestacc 57.7777777778 ccgbt 0.844444444444 bestgbt 0.888888888889
INFO:root:
Epoch: 142
INFO:root:ep 142 tracc 0.560175054705 lr 0.1 gbtacc 0.890590809628
INFO:root:Saving..
INFO:root:teacc 46.6666666667 bestacc 57.7777777778 ccgbt 0.844444444444 bestgbt 0.888888888889
INFO:root:
Epoch: 143
INFO:root:ep 143 tracc 0.560175054705 lr 0.1 gbtacc 0.890590809628
INFO:root:Saving..
INFO:root:teacc 46.6666666667 bestacc 57.7777777778 ccgbt 0.844444444444 bestgbt 0.888888888889`

Could you please share some time to answer my doubts? What can I do to solve this problem?
Additionally, the "CROPSIZE” is set as 17 in code but 32 in paper, did I misunderstand it?
The attachment of train log is:
[log-9.txt](https://github.com/wentaozhu/DeepLung/files/2231001/log-9.txt)

**Comment by wentaozhu** [comment]
You may use the paper as the golden standard. Learning rate, 0.1, is too high. Try to use 0.001.

**Comment by TracelessLe** [comment]
Thanks for responsor. I use the config in the paper as:

>The total number of epochs is
1,050. The initial learning rate is 0.01, and reduce to 0.001
after epoch 525, and finally to 0.0001 after epoch 840.

And I retrain the nodule classification network, the loss is still hold the line (~0.693). It's so werid.

New Log:
[log-9-new.txt](https://github.com/wentaozhu/DeepLung/files/2237609/log-9-new.txt)

I wonder if something wrong happend in the data process of the classification stage...

**Comment by wentaozhu** [comment]
Here is a guy obtaining better results inspired by our DeepLung https://github.com/uci-cbcl/DeepLung/issues/4

From the log, the GBM accuracy is high. But the 3D DPN accuracy is low. It is weird. Did you change your input size for 3D DPN to 33, and raw input feature for GBM as size 17?

**Comment by TracelessLe** [comment]
Thanks very much. I modified the mistakes in the 155th line and 71th line of the "extclsshpinfo.py". Then I change the input size for DPN to 32, GBM to 16. Now the loss descends and accuracy rises.
![1](https://user-images.githubusercontent.com/22371026/43381073-fb0770e0-9405-11e8-983c-926717795d41.png)

**Comment by wentaozhu** [comment]
Congratulations! You reproduced our results for the CAD system. Previously, several guys reproduced the detection system. Good luck on your future research!

**Comment by TracelessLe** [comment]
@anruoxi123 Hello, you need to modify the source code a little. The orginal code is not working smoothly.

**Comment by TracelessLe** [comment]
It's been too long. I can't remember clearly about the change. What I remember is that the method of combining the pid and srcid need to keep consistent in the whole time. The advice for you is to debug the code by code clock.

**Comment by anruoxi123** [comment]
Ok, thank you very much for your advice!
